### PR TITLE
[Asp.Net Core, HttpClient] Clarify Enrich callback sequence

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

Adding some clarification on sequencing. This is a commonly asked question/issue.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 1]
```diff
@@ -275,6 +275,34 @@ is the general extensibility point to add additional properties to any activity.
 The `Enrich` option is specific to this instrumentation, and is provided to
 get access to `HttpRequest` and `HttpResponse`.
 
+When overriding the default settings provided by instrumentation or adding
+additional telemetry, it is important to consider the sequence of callbacks.
+Generally, it is recommended to use `EnrichWithHttpResponse` for any activity
+enrichment that does not need access to exceptions, as the instrumentation
+library sets all defaults before this callback. The following is the sequence in
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:282]:
```suggestion
library populates telemetry using OTel sem conv, before this callback. The following is the sequence in
```

**Comment by cijothomas** [comment]
@vishweshbankwar Could you do a similar doc fix for Filter, to call out the Sampling vs Processor vs FilterCallback order?

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 2]
```diff
@@ -275,6 +275,34 @@ is the general extensibility point to add additional properties to any activity.
...
+which these callbacks are executed:
+
+1) Processor `OnStart`
+2) `EnrichWithHttpRequest`
+3) `EnrichWithException`
+4) `EnrichWithHttpResponse`
+5) Processor `OnEnd`
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:289]: nit: Would be nice to call out that `SetTag` will override any existing tag of same name, and `SetTag` with null is a way to achieve removing a Tag completely.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md:291]: I added in the example code.

**Comment by vishweshbankwar** [comment]
> @vishweshbankwar Could you do a similar doc fix for Filter, to call out the Sampling vs Processor vs FilterCallback order?

@cijothomas - We do clarify the sequence of filter versus sampler in the doc. For processor, I think its slightly more complex as that requires some explanation on what could happen to E2E trace if the span is filtered at later stage. I am thinking to clarify it here first: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/extending-the-sdk/MyFilteringProcessor.cs.

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 3]
```diff
@@ -275,6 +275,42 @@ is the general extensibility point to add additional properties to any activity.
...
+the library you can do so as follows:
+
+```csharp
+.AddAspNetCoreInstrumentation(o =>
+{
+  o.EnrichWithHttpResponse = (activity, response) =>
+  {
+      // Access request object if needed
+      // response.HttpContext.Request
+      activity.DisplayName = CustomDisplayName;
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:303]: nit
```suggestion
      activity.DisplayName = "CustomDisplayName";
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 4]
```diff
@@ -341,6 +341,44 @@ general extensibility point to add additional properties to any activity. The
...
+It is generally recommended to use `EnrichWithHttpResponseMessage` or
+`EnrichWithHttpWebResponse` for any activity enrichment that does not require
+access to exceptions or request object in case of .NET Framework, as the
+instrumentation library populates all telemetry following the [OTel
+specification](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/http)
+before this callback. The following is the sequence in which these callbacks are
+executed:
+
+1) Processor `OnStart`
+2) `EnrichWithHttpRequestMessage` (.NET) \ `EnrichWithHttpWebRequest` (.NET Framework)
```
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.Instrumentation.Http/README.md:356]: nit: The forward slash / is commonly used to indicate "or"
```suggestion
2) `EnrichWithHttpRequestMessage` (.NET) /  `EnrichWithHttpWebRequest` (.NET Framework)
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 5]
```diff
@@ -275,6 +275,42 @@ is the general extensibility point to add additional properties to any activity.
 The `Enrich` option is specific to this instrumentation, and is provided to
 get access to `HttpRequest` and `HttpResponse`.
 
+When overriding the default settings provided by instrumentation or adding
+additional telemetry, it is important to consider the sequence of callbacks.
+Generally, it is recommended to use `EnrichWithHttpResponse` for any activity
+enrichment that does not need access to exceptions, as the instrumentation
+library populates all telemetry following the [OTel
+specification](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/http)
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:283]:
```suggestion
library populates all telemetry following the [OpenTelemetry semantic
convention](https://github.com/open-telemetry/semantic-conventions/tree/v1.27.0/docs/http)
```

Same for the second readme

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md (hunk 5 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/README.md]: Thanks - updated

**Pull request merged**